src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -287,6 +290,33 @@ public IActionResult GetAdminToken()
             return Ok(requestHeaderToken);
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            // The function must be called warmup and should be of type warmup trigger
+            FunctionDescriptor function = scriptHost.GetFunctionOrNull(WarmUpConstants.FunctionName);
+            if (function != null && function.Metadata.ContainsInputBinding(WarmUpConstants.WarmupTrigger))
+            {
+                var hostStatus = new HostStatus
+                {
+                    State = scriptHostManager.State.ToString()
+                };
+
+                ParameterDescriptor inputParameter = function.Parameters.First(p => p.IsTrigger);
+                var arguments = new Dictionary<string, object>()
+                {
+                    { inputParameter.Name, JsonConvert.SerializeObject(hostStatus) }
```

Comment:
Once we move to WarmupContext, we won't serialize it - we'll pass the instance in directly

Reply:
Is this still planned?

Reply:
Yes - we should remove this serialization now.